### PR TITLE
Convert card module from directives to components

### PR DESCRIPTION
**Pull request by dgutride**
This completes the entire card module conversion - all unit tests work, ngdocs work correctly and all examples are building and working.

### CODE REVIEW

**Comment by dtaylor113** [comment]
Why is pfAggregateStatusCard marked as Deprecated?
image

**Comment by dgutride** [comment]
@dtaylor113 - the boolean/alt-layout is marked as deprecated, not the status card - this is not a change from the current behavior

**Comment by dtaylor113** [comment]
LGTM.  Minor issue, not introduced in this PR, but the card examples titles are sort of looking like params?  I don't think the @name likes dashes:

`@name patternfly.card.directive:pfCard - Timeframe Filters`

image

**Comment by dlabrecq** [comment]
LGTM. I'd merge it, but not sure there are any changes need in regard to @dtaylor113 's comment?

**Comment by dtaylor113** [comment]
My issues shouldn't hold up merging this PR :-)

File: src/card/basic/card-filter.html [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-  <span class="caret"></span>
-</button>
-<ul uib-dropdown-menu class="dropdown-menu-right" role="menu">
-  <li ng-repeat="item in filter.filters" ng-class="{'selected': item === currentFilter}">
-    <a role="menuitem" tabindex="-1" ng-click="filterCallBackFn(item)">
-      {{item.label}}
-    </a>
-  </li>
-</ul>
+<div uib-dropdown class="card-pf-time-frame-filter">
```
Comment (dgutride) [inline src/card/basic/card-filter.html]: Yeah - it actually came in from master (it was a contributed change after we branched for 4.0) - I ported it in so I wouldn't lose the change.

@dlabrecq

**Comment by dgutride** [comment]
@dlabrecq @dtaylor113  - I looked into those dashes in the name and we can't change that.  ngdocs uses the @name parameter to generate the left-hand navigation and if we remove those dashes, we'll end up with multiple entries in the navigation with the same name.

**Comment by dlabrecq** [comment]
Ok @dtaylor113, I'll merge.

**Pull request merged**